atomate/vasp/fireworks/exchange.py

```diff
@@ -0,0 +1,157 @@
+
+import numpy as np
+
+__author__ = "Nathan C. Frey"
+__email__ = "ncfrey@lbl.gov"
+
+
+class HeisenbergModelFW(Firework):
+    def __init__(
+        self,
+        exchange_wf_uuid,
+        parent_structure,
+        parents,
+        db_file=DB_FILE,
+        heisenberg_settings=None,
+        name="heisenberg model",
+        c=None,
+        structures=None,
+        energies=None,
+    ):
+        """
+        Takes a set of low-energy magnetic orderings and energies and maps them to a Heisenberg Model to compute exchange params.
+
+        Args:
+            exchange_wf_uuid (int): Unique id for record keeping.
+            parent_structure (Structure): Magnetic ground state.
+            parents (FireWorks): Parent FWs.
+            db_file (str): Path to file containing db credentials.
+            heisenberg_settings (dict): A config dict for Heisenberg model mapping.
+            name (str): Labels the FW.
+            c (dict): Config dict.
+            structures (list): Magnetic structures.
+            energies (list): Total energies of magnetic structures.
+
+        """
+
+        cutoff = heisenberg_settings["cutoff"]
+        tol = heisenberg_settings["tol"]
+        avg = heisenberg_settings["avg"]
+
+        fw_name = "%s %s" % (parent_structure.composition.reduced_formula, name)
+
+        additional_fields = {
+            "task_label": fw_name,
+            "exchange": {
+                "calc_type": name,
+                "wf_uuids": [],
+                "_source_wf_uuid": exchange_wf_uuid,
+            },
+        }
+
+        tasks = []
+
+        # Generate a HeisenbergModel with only <J> exchange
+        if avg:
+            tasks.append(
+                HeisenbergModelMapping(
+                    db_file=db_file,
+                    exchange_wf_uuid=exchange_wf_uuid,
+                    parent_structure=parent_structure,
+                    cutoff=cutoff,
+                    tol=tol,
+                    avg=avg,
+                    structures=structures,
+                    energies=energies,
+                )
+            )
+
+        else:
+            end_cutoff = 5.0  # Larger than this is unreasonable
+
+            for coff in np.linspace(cutoff, end_cutoff, 5):
+                tasks.append(
+                    HeisenbergModelMapping(
+                        db_file=db_file,
+                        exchange_wf_uuid=exchange_wf_uuid,
+                        parent_structure=parent_structure,
+                        cutoff=coff,
+                        tol=tol,
+                        avg=avg,
+                    )
+                )
+
+        super().__init__(tasks=tasks, name=fw_name, parents=parents)
+
+
+class VampireCallerFW(Firework):
+    def __init__(
+        self,
+        exchange_wf_uuid,
+        parent_structure,
+        parents,
+        db_file=DB_FILE,
+        mc_settings=None,
+        name="vampire caller",
+        c=None,
+        avg=True,
```

Comment:
Please rename to average throughout. We prefer explicit variable names where possible.

Reply:
Done!